tests/testthat/test-generate_data.R

```diff
@@ -669,23 +617,25 @@ test_that("genMultiFac works.", {
   # checks all values are in correct range
   expect_true(all(g3[, 2:nFac, with = FALSE] <= nLev && g3[, 2:nFac, with = FALSE] > 0))
 
-  # # checks each column has appropriate number of each value in it (1s, 2s, 3s, etc.)
-  # for(i in 2:(nFac + 1)) {
-  #   t <- table(g3[, i, with = FALSE])
-  #   sum_num <- (nLev^(nFac - 1))*nEach
-  # 
-  #   expect_true(all(t == sum_num))
-  #   expect_equal(sum(t), sum_num * nLev)
-  # }
-  
-  # sumTable <- table(rowSums(g3[, 2:length(g3)]))
-  # # checks all unique row sums are accounted for
-  # for(i in nFac:(nFac * nLev)) {
-  #   ind <- as.character(i)
-  #   expect_true(!is.na(sumTable[ind]))
-  # }
   
   ## levels == other, len(levels) != 1
+  nFac <- sample(2:5, size = 1)
+  nEach <- sample(2:5, size = 1)
+  nLev <- NULL
+  for(i in 1:nFac) {
+    nLev <- c(nLev, sample(2:5, size = 1))
+  }
+  g4 <- genMultiFac(nFac, each = nEach, levels = nLev)
+  
+  # checks all rows are unique
+  rowStrings <- unlist(lapply(split(g4[,-1], seq(nrow(g4))), function(x) paste0(x, collapse="")))
+  expect_true(length(rowStrings) == (length(unique(rowStrings)) * nEach))
+  
+  # checks there is right number of rows
+  expect_true(length(rowStrings) == nEach * prod(nLev))
+  
+  # checks all values are in correct range
+  expect_true(all(g4[, 2:nFac, with = FALSE] <= max(nLev) && g4[, 2:nFac, with = FALSE] > 0))
```

Comment:
Everytime I run this test (for all 4 scenarios), it throws an error, not sure what to do about it. It seems like it's an issue with all()? But I could be wrong. The warning for one of them is as follows:
── Warning (test-generate_data.R:618:3): genMultiFac works. ────────────────────
'length(x) = 5625 > 1' in coercion to 'logical(1)'
Backtrace:
 1. testthat::expect_true(...)
      at test-generate_data.R:618:2
 2. testthat::quasi_label(enquo(object), label, arg = "object")
 3. rlang::eval_bare(expr, quo_get_env(quo))

Is the vector being passed to all too long?

Reply:
Weird - I'm not getting any errors. But I do see a couple of flaws in the code. First, you need to look at columns `2 : (nFac + 1)`, or alternatively you need to look at `g4[, -1]`. But, even with that, you really need to evaluate each column separately, because (in this case), there is a maximum value that differs by column.

Reply:
Maybe it has something to do with my computer processing power, I don't have my desktop to compare it with. I fixed the (nFac + 1) problem in a handful of other places too, I had been making that mistake for a little while. I also addressed your second point.

Reply:
Is it still generating an error on your machine? It wouldn't have anything to do with processing power - it looked like some mismatched vector sizes, or like it was looking for a scalar. But it is weird that it was happening only on your machine. If it is still there, we need to get rid of the error.

Reply:
Yes I'm still getting it, they're warnings not errors, but that doesn't really make much of a difference

Reply:
That is so strange - I am not getting that - it all seems good. Try to break up the test into two pieces and see where it fails.

Reply:
Splitting it up seemed to fix it